### PR TITLE
feat: add relu composite

### PR DESCRIPTION
**Pull request by Miracle2333**
### PR types
New features

### PR changes
APIs

### Describe
add composite rule of relu

### CODE REVIEW

File: python/paddle/incubate/autograd/composite_rules.py [hunk 1]
```diff
@@ -223,3 +223,10 @@ def bernoulli(shape, dtype, p, seed=0):
         ),
         dtype,
     )
+
+
+@REGISTER_COMPOSITE('relu')
+def relu_composite(x):
+    """define composite rule of op relu"""
+    # # /* relu(x) = max(x, 0) */ copy from relu-kernel.cc
```
Comment (cyber-pioneer) [inline python/paddle/incubate/autograd/composite_rules.py]: polish comment

Comment (Miracle2333) [reply python/paddle/incubate/autograd/composite_rules.py:231]: 已经修改了

Comment (Miracle2333) [reply python/paddle/incubate/autograd/composite_rules.py]: done

File: python/paddle/incubate/autograd/composite_rules.py [hunk 2]
```diff
@@ -223,3 +223,10 @@ def bernoulli(shape, dtype, p, seed=0):
         ),
         dtype,
     )
+
+
+@REGISTER_COMPOSITE('relu')
+def relu_composite(x):
+    """define composite rule of op relu"""
+    # relu(x) = max(x, 0)
+    return maximum(0, x)
```
Comment (JiabinYang) [inline python/paddle/incubate/autograd/composite_rules.py]: using commit hook?

File: python/paddle/fluid/tests/unittests/test_activation_op.py [hunk 3]
```diff
@@ -1540,8 +1540,10 @@ def init_shape(self):
 class TestRelu(TestActivation):
     def setUp(self):
         self.op_type = "relu"
+        self.prim_op_type = "comp"
```
Comment (JiabinYang) [inline python/paddle/fluid/tests/unittests/test_activation_op.py:1779]: no python api used?

**Review by JiabinYang — COMMENTED** [review]
some comments

File: python/paddle/fluid/tests/unittests/test_activation_op.py (hunk 3 above)
Comment (Miracle2333) [reply python/paddle/fluid/tests/unittests/test_activation_op.py:1779]: python api of relu added

File: python/paddle/incubate/autograd/composite_rules.py (hunk 2 above)
Comment (Miracle2333) [reply python/paddle/incubate/autograd/composite_rules.py]: done

**Review by JiabinYang — DISMISSED** [review]
LGTM

File: python/paddle/incubate/autograd/composite_rules.py (hunk 2 above)
Comment (Miracle2333) [reply python/paddle/incubate/autograd/composite_rules.py]: image

**Review by cyber-pioneer — DISMISSED** [review]
LGTM

**Review by cyber-pioneer — DISMISSED** [review]
LGTM

**Review by Charles-hit — COMMENTED** [review]
relu fp16单测请打开组合测试

**Review by cyber-pioneer — APPROVED** [review]

**Review by Charles-hit — APPROVED** [review]

**Pull request merged**